python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_base.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from azure.cosmos.aio import ContainerProxy, CosmosClient, DatabaseProxy
+from azure.cosmos.exceptions import CosmosResourceNotFoundError
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_settings import AzureCosmosDBNoSQLSettings
+from semantic_kernel.connectors.memory.azure_cosmos_db.utils import CosmosClientWrapper, DefaultAzureCredentialWrapper
+from semantic_kernel.exceptions.memory_connector_exceptions import (
+    MemoryConnectorInitializationError,
+    MemoryConnectorResourceNotFound,
+)
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+
+@experimental_class
+class AzureCosmosDBNoSQLBase(KernelBaseModel):
+    """An Azure Cosmos DB NoSQL collection stores documents in a Azure Cosmos DB NoSQL account."""
+
+    cosmos_client: CosmosClient
+    database_name: str
+    # If create_database is True, the database will be created
+    # if it does not exist when an operation requires a database.
+    create_database: bool
+
+    def __init__(
+        self,
+        database_name: str,
+        url: str | None = None,
+        key: str | None = None,
+        cosmos_client: CosmosClient | None = None,
+        create_database: bool = False,
+        **kwargs,
```

Comment:
let's make the managed_client arg explicit.

Reply:
This is not a derived class from vector store or vector collection.